sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -120,9 +122,24 @@ struct _pi_device : _pi_object {
   ze_device_compute_properties_t ZeDeviceComputeProperties;
 };
 
+// Implements memory allocation via L0 RT for USM allocator interface.
+class USMSharedMemoryAlloc: public SystemMemory {
+private:
+  // The plugin has exactly one pi_device per pi_context object.
```

Comment:
You should not base your design on this (temporary) fact.

Reply:
Sergey is correct - that's an assumption that goes against desired usage models.